spark/src/main/scala/geotrellis/spark/io/cog/COGLayerWriter.scala

```diff
@@ -50,32 +53,30 @@ trait COGLayerWriter extends LazyLogging with Serializable {
     K: SpatialComponent: Ordering: JsonFormat: ClassTag,
     V <: CellGrid: ClassTag: ? => TileMergeMethods[V]: ? => TilePrototypeMethods[V]: ? => TileCropMethods[V]: GeoTiffReader: GeoTiffBuilder
   ](
-     layerName: String,
+     id: ID,
      tiles: RDD[(K, V)] with Metadata[TileLayerMetadata[K]],
-     tileZoom: Int,
      keyIndexMethod: KeyIndexMethod[K]
-   ): Unit = write[K, V](layerName, tiles, tileZoom, keyIndexMethod, NoCompression, None)
+   ): Unit = write[K, V](id, tiles, keyIndexMethod, NoCompression, None)
 
   def write[
     K: SpatialComponent: Ordering: JsonFormat: ClassTag,
     V <: CellGrid: ClassTag: ? => TileMergeMethods[V]: ? => TilePrototypeMethods[V]: ? => TileCropMethods[V]: GeoTiffReader: GeoTiffBuilder
   ](
-    layerName: String,
+    id: ID,
     tiles: RDD[(K, V)] with Metadata[TileLayerMetadata[K]],
-    tileZoom: Int,
     keyIndexMethod: KeyIndexMethod[K],
     compression: Compression,
     mergeFunc: Option[(GeoTiff[V], GeoTiff[V]) => GeoTiff[V]]
   ): Unit =
     tiles.metadata.bounds match {
       case keyBounds: KeyBounds[K] =>
-        val cogLayer = COGLayer.fromLayerRDD(tiles, tileZoom, compression = compression)
+        val cogLayer = COGLayer.fromLayerRDD(tiles, id.asInstanceOf[LayerId].zoom, compression = compression)
```

Comment:
Eh, `asInstanceOf[LayerId]` kills the idea of having a generic `ID` in the interface. It looks like in this case we can have only `LayerId` (?)

Would not such API be a bit confusing for the user?

Reply:
Yeah, I was thinking that it might be better to just use `LayerId` instead of `ID`. Which API do you mean? The one that uses `LayerId`?

Reply:
Yeah, this move to `LayerId` changes semantics a bit; because COG layer always builds a pyramid right now

Reply:
Do you think it'd be better to go back to having the `layerName` and `layerZoom` parameters for the `write` methods?